Sources/Sentry/SentryOptions.m

```diff
@@ -235,12 +240,17 @@ - (BOOL)validateOptions:(NSDictionary<NSString *, id> *)options
         }
     }
 
-    NSString *dsn = @"";
-    if (nil != options[@"dsn"] && [options[@"dsn"] isKindOfClass:[NSString class]]) {
-        dsn = options[@"dsn"];
-    }
+    if (options[@"dsn"] != [NSNull null]) {
+        NSString *dsn = @"";
+        if (nil != options[@"dsn"] && [options[@"dsn"] isKindOfClass:[NSString class]]) {
+            dsn = options[@"dsn"];
+        }
 
-    self.parsedDsn = [[SentryDsn alloc] initWithString:dsn didFailWithError:error];
+        self.parsedDsn = [[SentryDsn alloc] initWithString:dsn didFailWithError:error];
+        if (self.parsedDsn == nil) {
+            return NO;
+        }
+    }
```

Comment:
I'm trying to understand why the test case flips from asserting the SDN is not nil to asserting that it is nil in the default case. Do we allow initializing the SDK without a valid DSN? If not, I think it would be clearer what the eventual outcome would be here if instead, we made this an early return like so:
```objective-c
if (options[@"dsn"] == [NSNull null]) {
    return NO;
}
```
and then do the rest of the work at one level higher of scope below that. It would also avoid all the work below being done which wouldn't matter if the SDK winds up in an error state anyways.

Reply:
The test case flipped because the prior assumption was wrong.

Why we allow options to be initialized without a DSN I dont know , maybe @philipphofmann can give us some context here.

Reply:
You can initialize the SDK without a DSN. The SDK would run and do its job, but the client will drop everything see https://github.com/getsentry/sentry-cocoa/blob/72c650a248ec6381ac4233a1dcb4d02b94bf65d6/Sources/Sentry/SentryClient.m#L636-L639

It's similar to using `options.enabled = false`.

Reply:
I opened an issue where we can discuss this further, since it's out of scope for this PR: https://github.com/getsentry/sentry-cocoa/issues/2712